client/client-multi/client-objective-c/Kaa/channel/impl/channels/DefaultOperationTcpChannel.m

```diff
@@ -281,9 +281,11 @@ - (void)stream:(NSStream *)aStream handleEvent:(NSStreamEvent)eventCode {
             __weak typeof(self) weakSelf = self;
             [self.executor addOperationWithBlock:^{
                 [weakSelf sendConnect];
-                [weakSelf schedulePingTask];
             }];
-            
+            dispatch_time_t time =  dispatch_time(DISPATCH_TIME_NOW, (int64_t)PING_TIMEOUT_SEC * NSEC_PER_SEC);
```

Comment:
Your doubled the ping timeout period by dispatching PingTask after delay (1st time), which uses that delay inside (2nd time).